tap_jira/streams.py

```diff
@@ -4,11 +4,48 @@
 import dateparser
 
 from singer import metrics, utils, metadata, Transformer
+from singer.transform import SchemaMismatch
+from dateutil.parser._parser import ParserError
 from .http import Paginator,JiraNotFoundError
 from .context import Context
 
 DEFAULT_PAGE_SIZE = 50
 
+def handle_date_time_schema_miss_match(exception, record): # pylint: disable=inconsistent-return-statements
+    """
+    Handling exception for date-time value out of range.
+    """
+
+    if ("{'format': 'date-time', 'type': ['string', 'null']}"
+        in exception.args[0].split("\n\t")[1]):
+        nested_keys = exception.args[0].split("\n\t")[1].split(":")[0]
+
+        obj = record.copy()
+        # Getting the date value from nested object
+        for key in nested_keys.split("."):
+            # Convert list index of nested object to integer
+            obj = obj[int(key) if key.isnumeric() else key]
+
+        try:
+            # Parsing date to catch 'out of range' error
+            utils.strptime_to_utc(obj)
+        except ParserError as err:
+
+            # Check the error message if the 'year' or 'day' is out of range
+            # example: year 51502 is out of range: 51502-06-08T14:46:42.000000
+            if ("out of range" in str(err)) or (
+                # Check the error message if 'month' or ['hours','minutes','seconds'] given in date is not in range
+                # example: month must be in 1..12: 5150-33-08T14:46:42.000000
+                "must be in" in str(err)):
+                LOGGER.warning("Skipping record of id: %s due to Date out of range, DATE: %s", record["id"], obj)
```

Comment:
The code will throw "KEYERROR", for the streams which do not contain id field

Reply:
Updated the code and used the id dynamically instead of static values.